src/zoom.js

```diff
@@ -345,7 +337,14 @@ export default function() {
       else if (g.touch1 && g.touch1[2] === t.identifier) delete g.touch1;
     }
     if (g.touch1 && !g.touch0) g.touch0 = g.touch1, delete g.touch1;
-    if (!g.touch0) g.end();
+    if (!g.touch0) {
+      g.end();
+      // If this was a dbltap, reroute to the (optional) dblclick.zoom handler.
+      if (g.consecutive && g.tap) {
+        var p = select(this).on("dblclick.zoom");
+        if (p) p.apply(this, arguments);
+      }
+    }
```

Comment:
perhaps it would be better to simply `select(this).dispatch("dblclick")`?

Reply:
No, because that would notify all dblclick listeners, not just trigger the zoom transition, and those listeners would receive a CustomEvent rather than a proper MouseEvent.